streaming/src/main/scala/org/apache/spark/streaming/dstream/InputDStream.scala

```diff
@@ -47,6 +49,27 @@ abstract class InputDStream[T: ClassTag] (@transient ssc_ : StreamingContext)
   /** This is an unique identifier for the input stream. */
   val id = ssc.getNewInputStreamId()
 
+  /**
+   * A rate estimator configured by the user to compute a dynamic ingestion bound for this stream.
+   * @see `RateEstimator`
+   */
+  protected [streaming] val rateEstimator = newEstimator()
```

Comment:
Why do you need `rateEstimator` as a field in `DStream`? All you need expose to modules outside DStream is the RateController. And the RateEstimator can be within the `RateController`